### PR TITLE
aded files

### PR DESCRIPTION
**Pull request by marsermd**
Панин Михаил 2539

### CODE REVIEW

**Comment by thevery** [comment]
app icon is missing

**Comment by thevery** [comment]
some images are not updated until scroll (it seems that you're missing notifyDatasetChanged())

**Comment by thevery** [comment]
no top/bottom padding

**Comment by thevery** [comment]
big image should load eagerly, not lazily

**Comment by thevery** [comment]
empty dialog on loading is ugly

**Comment by thevery** [comment]
offline is not working

**Comment by thevery** [comment]
no refresh button

**Comment by thevery** [comment]
app is ok (15-4=11), code review pending

File: res/layout/image_actitvity.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout/image_actitvity.xml:3]: useless layout

File: src/ru/marsermd/flikrVigintus/ImageActivity.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * Time: 5:20
+ * To change this template use File | Settings | File Templates.
+ */
+public class ImageActivity extends Activity {
+    private static SimpleCachingDownloadTask downloadTask;
+    private ImageView fullImage;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        Log.e("wow!", "started!!!!!!!!!!!!!!!!!!!1");
```
Comment (thevery) [inline src/ru/marsermd/flikrVigintus/ImageActivity.java:27]: bad logs

File: src/ru/marsermd/flikrVigintus/ImageActivity.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        Bundle extra = getIntent().getExtras();
+
+        if (downloadTask != null)
+            downloadTask.cancel(true);
+
+        fullImage = (ImageView) findViewById(R.id.fullscreen_image);
+
+
+        downloadTask = new SimpleCachingDownloadTask(extra.getString("ImageUrl")) {
```
Comment (thevery) [inline src/ru/marsermd/flikrVigintus/ImageActivity.java:39]: `"ImageUrl"` should be const

File: src/ru/marsermd/flikrVigintus/SearchActivity.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    private void init() {
+        FileCacher.getInstance().init(this, CACHE_BYTE_LIMIT);
+
+        searchedImages = (ListView) findViewById(R.id.images_result);
+
+        imm = (InputMethodManager) getSystemService(getBaseContext().INPUT_METHOD_SERVICE);
+
+        Display display = ((WindowManager) getSystemService(WINDOW_SERVICE)).getDefaultDisplay();
+        int orientation = display.getRotation();
+        if (orientation == Surface.ROTATION_90 || orientation == Surface.ROTATION_270) {    // landscape
```
Comment (thevery) [inline src/ru/marsermd/flikrVigintus/SearchActivity.java:48]: use `layout` (or `values`) and `layout-land` (`values-land`) resource folders instead

File: res/layout-land/rowlayout.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout-land/rowlayout.xml:1]: why not just GridView

File: src/ru/marsermd/flikrVigintus/utility/http/translate/yandex/YandexTranslateTask.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
+package ru.marsermd.flikrVigintus.utility.http.translate.yandex;
```
Comment (thevery) [inline src/ru/marsermd/flikrVigintus/utility/http/translate/yandex/YandexTranslateTask.java:1]: unused

**Comment by thevery** [comment]
executors overuse and overengineering at all

**Comment by thevery** [comment]
15-4-4=7

File: src/ru/marsermd/flikrVigintus/utility/http/translate/yandex/YandexTranslateTask.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import ru.marsermd.flikrVigintus.utility.http.translate.TranslateResult;
+import ru.marsermd.flikrVigintus.utility.http.translate.TranslateTask;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Hashtable;
+import java.util.Map;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: georgeee
```
Comment (thevery) [inline src/ru/marsermd/flikrVigintus/utility/http/translate/yandex/YandexTranslateTask.java:15]: orly? ;)

Comment (marsermd) [inline src/ru/marsermd/flikrVigintus/utility/http/translate/yandex/YandexTranslateTask.java:15]: yep. Copy&pasted code from our work in pairs called "silencio". As long as i remember, we are allowed to re-use our old code.

Comment (thevery) [inline src/ru/marsermd/flikrVigintus/utility/http/translate/yandex/YandexTranslateTask.java:15]: just kidding - check the smile ;)

**Pull request closed without merging**